doc.go

```diff
@@ -0,0 +1,89 @@
+// Generated by running
+//  go generate github.com/gonum/matrix
+// DO NOT EDIT.
+
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package matrix provides common error handling mechanisms for matrix operations
+// in mat64 and cmat128.
+//
+// Overview
+//
+// This section provides a quick overview of the matrix package. The following
+// sections provide more in depth commentary.
+//
+// matrix provides:
+//  - Error type definitions
+//  - Error recovery mechanisms
+//  - Common constants used by mat64 and cmat128
+//
+// Errors
+//
+// The mat64 and cmat128 matrix packages share a common set of errors
+// provided by matrix via the matrix.Error type. Errors are either
+// returned in cases where the function may meaningfully fail such as
+// the mat64 solve methods, or may be the parameter for a panic call
+// when there is a mismatch between the provided parameter and the
```

Comment:
Okay. I don't think the use of the word "parameter" as both the input to panic and input to function. Maybe just call "input" the thing provided to panic, i.e. "may be the input to a panic call when there is a mismatch between the provided parameter(s) and ..."